### PR TITLE
det-975: get probe sleep interval from BE

### PR DESCRIPTION
**Pull request by makk94**
Also, fix upgrade path for raindrop :( 

If when this goes in, I'll update probe_tasks probe version as well to trigger upgrades (for nocturnal anyway)

PR bundle:
- https://github.com/preludeorg/backbone/pull/719
- https://github.com/preludeorg/service/pull/1601
- https://github.com/preludeorg/libraries/pull/934

### CODE REVIEW

**Review by packplusplus — COMMENTED** [review]
I don't love it. 

In the case of stop or upgrade, the agent was saying "did i get the value stop, if so exit"

In this case we're just grabbing the value, not sanity checking it, and using it. 

So the nocturnal one could 100% just be like "5; sudo reboot" and I think it would execute that code. Not sure about the powershell one, but similar concern there.

**Comment by makk94** [comment]
@packplusplus you have not been requested to review this PR

**Comment by packplusplus** [comment]
*cough* draft *cough*

File: shell/probe/nocturnal.sh [hunk 1]
```diff
@@ -18,6 +18,12 @@ do
     else
         rm -rf $vst
         unset dat
-        sleep 3600
+
+        # Option 1. Explicitly check that sleep is an int. Otherwise default to 1800s
```
Comment (bfuzzy1) [inline shell/probe/nocturnal.sh:22]: I am a fan of the explicity you have in option 1. I would approve that option.

Comment (makk94) [reply shell/probe/nocturnal.sh]: kewl. patrick agrees

out of curiosity, can you think of a way to take advantage of option 2?

Comment (bfuzzy1) [reply shell/probe/nocturnal.sh]: I honestly can not think of anything worth the effort.

**Review by packplusplus — APPROVED** [review]
I'm convinced. Thank you for sanitizing your inputs.

**Pull request merged**